doc/developer/tracing.md

```diff
@@ -0,0 +1,231 @@
+# Introduction
+
+[`tracing`] is a set of crates used for _logging_ and _tracing_ of rust
+binaries. It is primarily composed of the [`tracing`] crate, which is the
+core way to interact with the framework, and the [`tracing-subscriber`] crate,
+which provides an API to compose systems that consume tracing data.
```

Comment:
I think it'd be great to explain what tracing *is* and why it is not logging. We can probably just copy the explanation from `tracing` crate and credit it appropriately:

> In asynchronous systems like Tokio, interpreting traditional log messages can often be quite challenging. Since individual tasks are multiplexed on the same thread, associated events and log lines are intermixed making it difficult to trace the logic flow. tracing expands upon logging-style diagnostics by allowing libraries and applications to record structured events with additional information about temporality and causality — unlike a log message, a span in tracing has a beginning and end time, may be entered and exited by the flow of execution, and may exist within a nested tree of similar spans. In addition, tracing spans are structured, with the ability to record typed data as well as textual messages.
> —https://docs.rs/tracing/latest/tracing/#overview

Reply:
Which is extra confusing because we also use `tracing` for logging!